batch/batch/driver/job.py

```diff
@@ -244,7 +244,7 @@ async def make_request():
             await instance.incr_failed_request_count()
             raise
 
-    await retry_transient_errors(make_request)
+    await retry_transient_errors_n_times(5, make_request)
```

Comment:
If we reach the maximum number of failures, what happens? Will the worker eventually learn that the job is deleted?

Reply:
At least this way, we'll get an interrupt. But this job will tried to be unscheduled over and over again from the SQL query until the worker is gone.